lara-typescript/src/section-authoring/hooks/use-api-provider.ts

```diff
@@ -125,38 +125,31 @@ export const usePageAPI = () => {
 
   const currentPage = getPage();
 
-  const moveSection = (
-    sectionId: string,
-    selectedPageId: string,
-    relativePosition: "before" | "after",
-    selectedOtherSectionId: string
-    ) => {
-
-    const section = getSection(sectionId);
-    if (!section) return;
-    const sections = getSections();
-    const sectionIndex = sections.findIndex(s => s.id === sectionId);
-    const otherSectionIndex = sections.findIndex(s => s.id === selectedOtherSectionId);
-    const otherSection = getSections()[otherSectionIndex];
-    const newIndex = otherSectionIndex
-                        ? relativePosition === "after"
-                          ? otherSectionIndex + 1
-                          : otherSectionIndex - 1
-                        : 0;
-    const updatedSections = getSections();
-    updatedSections.splice(sectionIndex, 1);
-    updatedSections.splice(newIndex, 0, section);
-    let sectionsCount = 0;
-    updatedSections.forEach((s, index) => {
-      if (otherSection) {
-        updatedSections[index].position = ++sectionsCount;
+  const moveSection = (sectionId: string, destination: ISectionDestination) => {
+    if (currentPage) {
+      const changes = _moveSection({sectionId, destination, pages: getPages.data || []});
+      if (changes) {
+        updateSectionsMutation.mutate({...changes[0]});
       }
-    });
-    if (updateSections && updatedSections) {
-      updateSections({id: selectedPageId, sections: updatedSections});
+    }
+    else {
+      // tslint:disable-next-line
+      console.error("no page specified, cant invoke method.");
```

Comment:
Should this be surfaced to the user as an alert?

Reply:
That is a good question. I think for now I will leave it.